charts/streams-app/templates/deployment.yaml

```diff
@@ -48,17 +48,17 @@ spec:
         prometheus.io/scrape: "true"
         prometheus.io/port: {{ .Values.prometheus.jmx.port | quote }}
       {{- end }}
-      {{- range $key, $value := .Values.annotations }}
+      {{- range $key, $value := .Values.podAnnotations }}
         {{ $key | quote }}: {{ $value | quote }}
       {{- end }}
-      {{- if and .Values.autoscaling.consumerGroup (not .Values.annotations.consumerGroup) }}
+      {{- if and .Values.autoscaling.consumerGroup (not .Values.podAnnotations.consumerGroup) }}
```

Comment:
We don't actually need the consumer group annotation on the pod, only on the deployment, so you can remove it here